src/algorithms/GenericFunctions.jl

```diff
@@ -301,18 +301,85 @@ function gcdinv(a::T, b::T) where T <: RingElem
    return (g, s)
 end
 
+function _crt_with_lcm_stub(r1::T, m1::T, r2::T, m2::T; check::Bool=true) where T <: RingElement
+   diff = r2 - r1
+   if iszero(m1)
+      check && !is_divisible_by(diff, m2) && error("no crt solution")
+      return (r1, m1)
+   elseif iszero(m2)
+      check && !is_divisible_by(diff, m1) && error("no crt solution")
+      return (r2, m2)
+   end
+   g, s = gcdinv(m1, m2)
+   if isone(g)
+      return (r1 + mulmod(diff, s, m2)*m1, m1*m2)
+   elseif !check
+      m1og = divexact(m1, g; check=false)
+      return (r1 + mulmod(diff, s, m2)*m1og, m1og*m2)
+   else
+      m2og = divexact(m2, g; check=false)
+      diff = divexact(diff, g; check=check)
+      return (r1 + mulmod(diff, s, m2og)*m1, m1*m2og)
+   end
+end
+
+function _crt_stub(r1::T, m1::T, r2::T, m2::T; check::Bool=true) where T <: RingElement
+    return _crt_with_lcm_stub(r1, m1, r2, m2; check=check)[1]
+end
+
+@doc Markdown.doc"""
+    crt(r1::T, m1::T, r1::T, m2::T; check::Bool=true) where T <: RingElement
 
-# TODO: Move from CRT from Hecke to AbstractAlgebra?
```

Comment:
Why remove this comment? It is still missing the version using the tree.

Reply:
??
There is no crt_env here

Reply:
How julia does the "tree" is implementation defined

Reply:
The CRT stuff from Hecke is not moved to AA, so why remove the comment?

Reply:
Will this always be correct for Int? At least if product of moduli < typemax?

Reply:
ok